app/uk/gov/hmrc/alcoholdutyreturns/service/ReturnsService.scala

```diff
@@ -71,6 +74,42 @@ class ReturnsService @Inject() (
                                       returnId.appaId
                                     )
       _                          <- EitherT.right[ErrorResponse](cacheRepository.clearUserAnswersById(returnId))
-    } yield returnCreatedDetails
+    } yield {
+      cacheRepository.get(returnId).map {
+        case Some(ua) =>
+          auditReturnSubmitted(
+            ua,
+            returnToSubmit,
+            AdrReturnCreatedDetails.fromReturnCreatedDetails(returnCreatedDetails),
+            returnId
+          )
+        case None     =>
+          logger.warn("User answers couldn't be retrieved while auditing return submission")
```

Comment:
Maybe we shouldn't stop the whole audit event just because we couldn't receive user answers. I think the fields should be none or empty string if we can't get then fron useranswers